keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/DependsOnConstraintEvaluator.kt

```diff
@@ -18,9 +20,10 @@ class DependsOnConstraintEvaluator(
     artifact: DeliveryArtifact,
     version: String,
     deliveryConfig: DeliveryConfig,
-    targetEnvironment: String
+    targetEnvironment: Environment,
+    priorState: ConstraintState?
   ): Boolean {
-    val constraint = getConstraintForEnvironment(deliveryConfig, targetEnvironment, constraintType)
+    val constraint = getConstraintForEnvironment(deliveryConfig, targetEnvironment.name, constraintType)
```

Comment:
My feeling on simplifying this is that we should have a way to get `DeliveryConfig` from `Environment` (i.e. `@JsonBackReference` or whatever that thing is), and then just pass the `Environment` object in here. Outside of this change, tho.